hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/AbstractMessageTask.java

```diff
@@ -114,8 +115,18 @@ protected boolean requiresAuthentication() {
         return true;
     }
 
+    /**
+     * Used to accept hot restart messages (and some other messages required for
+     * client to connect) sent from MC client when node start is not complete yet.
+     */
+    protected boolean acceptOnIncompleteStart() {
+        return false;
+    }
+
     private void initializeAndProcessMessage() throws Throwable {
-        if (!node.getNodeExtension().isStartCompleted()) {
+        boolean acceptOnIncompleteStart = acceptOnIncompleteStart()
+                && ConnectionType.MC_JAVA_CLIENT.equals(endpoint.getClientType());
```

Comment:
There is a problem with this check for `AuthenticationMessageTask`. For this message `endpoint.getClientType()` returns `NONE`, as the client did not authenticate yet.

@sancar do you have any ideas for a workaround?

Reply:
I think I have an idea. Going to fix `testAuthentication_with_mcModeEnabled_when_clusterStart_isNotComplete` test first, so it starts failing and try implementing the workaround.